utils/forest_utils/src/db/file_backed_obj.rs

```diff
@@ -1,36 +1,61 @@
 // Copyright 2019-2023 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-use std::{path::PathBuf, str::FromStr};
+use std::{
+    path::PathBuf,
+    str::FromStr,
+    time::{Duration, SystemTime},
+};
 
+use ahash::HashSet;
 use cid::Cid;
 use log::warn;
 
 pub struct FileBacked<T: FileBackedObject> {
     inner: T,
     path: PathBuf,
+    last_sync: Option<SystemTime>,
 }
 
+const SYNC_PERIOD: Duration = Duration::from_secs(600);
+
 impl<T: FileBackedObject> FileBacked<T> {
     /// Gets a borrow of the inner object
     pub fn inner(&self) -> &T {
         &self.inner
     }
 
-    /// Gets a mutable borrow of the inner object
-    pub fn inner_mut(&mut self) -> &mut T {
-        &mut self.inner
-    }
-
     /// Sets the inner object and flushes to file
     pub fn set_inner(&mut self, inner: T) -> anyhow::Result<()> {
         self.inner = inner;
         self.flush_to_file()
     }
 
+    /// Calls func with inner mutable reference and flushes every `SYNC_PERIOD`
+    pub fn with_inner<F>(&mut self, func: F) -> anyhow::Result<()>
+    where
+        F: FnOnce(&mut T),
+    {
+        func(&mut self.inner);
+        let now = SystemTime::now();
+        if let Some(last_sync) = self.last_sync {
+            if now.duration_since(last_sync)? > SYNC_PERIOD {
+                self.last_sync = Some(now);
+                self.flush_to_file()?;
+            }
+            return Ok(());
+        }
+        self.last_sync = Some(now);
+        Ok(())
+    }
```

Comment:
Fancy adding a unit test for this method?

Reply:
Let me think.

Reply:
Added two tests.